go/pkg/client/retries_test.go

```diff
@@ -285,6 +285,36 @@ func TestWriteRetries(t *testing.T) {
 	}
 }
 
+func TestWriteRetriesWithOptions(t *testing.T) {
+	opts := [][]client.ByteStreamWriteOption{
+		{client.ByteSteramOptFinishWrite(true), client.ByteStreamOptOffset(0)},
+		{client.ByteSteramOptFinishWrite(false), client.ByteStreamOptOffset(0)},
+		{client.ByteSteramOptFinishWrite(true)},
+		{client.ByteSteramOptFinishWrite(false)},
+		{client.ByteStreamOptOffset(0)},
```

Comment:
Can you add a test case with a non-zero offset?

Reply:
Done.